src/obb/server/http_component.clj

```diff
@@ -0,0 +1,39 @@
+(ns ^{:added "0.1.0" :author "Pedro Pereira Santos"}
+  obb.server.http-component
+  "The aleph server as a component"
+  (:require [com.stuartsierra.component :as component]
+            [ring.middleware.cors :as cors]
+            [aleph.http :as http]
+            [compojure.core :as compojure]
+            [obb.server.routes :as routes]))
+
+(defn- setup-cors
+  "Setup cors"
+  [handler]
+  (cors/wrap-cors handler
+                  :access-control-allow-origin
+                  [#"^http://localhost(.*)"]
+                  :access-control-allow-methods [:get :put :post :delete]))
+
+(def app
+  "The main app handler"
+  (-> (compojure/routes routes/public-routes)
+      (setup-cors)))
+
+(defrecord HttpServerComponent [server port]
+
+  component/Lifecycle
+
+  (start [component]
+    (println (str "** OBB HTTP Server " (or (get (System/getenv) "OBB_ENV" "development"))
```

Comment:
I use `environ` from the beginning and provide a development configuration in `profiles.clj`. It is a bit annoying but that's the downside of making the configuration mandatory. To make it easier to set up, I normally do the following:
* Add a `profiles.example.clj` file to the repo, stating all the configurations for development and testing;
* Add a section to the README, providing instructions on how to set up the project. In that section I recommend that people copy `profiles.example.clj` to `profiles.clj` and only then run the tests / start the server locally.